dashboard/app/controllers/programming_environments_controller.rb

```diff
@@ -56,11 +56,20 @@ def update
   end
 
   def show
-    @programming_environment = ProgrammingEnvironment.find_by_name(params[:name])
     return render :not_found unless @programming_environment
     @programming_environment_categories = @programming_environment.categories.select {|c| c.programming_expressions.count > 0}.map(&:summarize_for_environment_show)
   end
 
+  def destroy
+    return render :not_found unless @programming_environment
+    if @programming_environment.destroy
+      File.delete(@programming_environment.file_path) if File.exist?(@programming_environment.file_path)
+      render(status: 200, plain: "Destroyed #{@programming_environment.name}")
+    else
+      render(status: :not_acceptable, plain: @programming_environment.errors.full_messages.join('. '))
```

Comment:
(minor, up to you) This is a bit pedantic, but I think "406 Not Acceptable" is a fairly specific error code around the content-type of the response.  If this error is likely to be a server-side error, then 500 is probably more appropriate?